tensorboard/compat/tensorflow_stub/io/gfile.py

```diff
@@ -19,7 +19,7 @@
 TensorBoard.  This allows running TensorBoard without depending on
 TensorFlow for file operations.
 """
-
+from adlfs import AzureBlobFileSystem
```

Comment:
We don't have adlfs libraries installed on development or ci machines so this is not going to build. You can see this in the failures for the build checks:

https://github.com/tensorflow/tensorboard/actions/runs/9352527415/job/25755552731?pr=6860

What would be a good way to check for this without importing AzureBlobFileSystem?

Reply:
Hi, if a file is initially created by opening it in append mode, this file will allow append operations later on. However, as shown in the code snippet above, the file is always created by opening it in write mode: 
```

        # write the first chunk to truncate file if it already exists
        self.fs.write(self.filename, file_content, self.binary_mode)
```

One other option, besides checking what filesystem we are dealing with, is to always open the file in append mode initially, when the fs supports that. However, this means the contents of the file, if it exists, will not be cleared.